Makefile

```diff
@@ -85,23 +91,42 @@ e2e-tests:
 
 
 e2e-tests-diagnostic:
-	$(PYTHON) $(TEST)/integration/run_tests.py --local --verbose
+	. $(VENV)/bin/activate; \
+	$(PYTHON) $(TEST)/integration/run_tests.py --local --verbose; \
 
 
 # run unit and CLI tests
 tests: unit-tests e2e-tests
 
 
+check-flake8:
+	. $(VENV)/bin/activate; \
+	echo "Flake8 linting for invalid source (bad syntax, undefined variables)..."; \
+	$(PYTHON) -m flake8 --count --select=E9,F63,F7,F82 --show-source --statistics; \
+	echo "Flake8 linting failure for CI..."; \
+	$(PYTHON) -m flake8 . --count --max-complexity=15 --statistics --max-line-length=127; \
+
+
+check-black:
+	. $(VENV)/bin/activate; \
+	$(PYTHON) -m black --check --target-version py36 $(MRLN); \
+
+
+check-pylint:
+	. $(VENV)/bin/activate; \
+	echo "PyLinting merlin source..."; \
+	$(PYTHON) -m pylint merlin --rcfile=setup.cfg --ignore-patterns="$(VENV)/" --disable=logging-fstring-interpolation; \
```

Comment:
As a note on what this is doing, it is: ignoring PyLint examination of the venv (we don't care if our dependencies are linted); ignoring f-string interpolation.

Pylint warns about using f-strings in Loggers because Loggers have a particular optimization about interpolating % strings (namely, it does not evaluate the string expansion fr a % str if a logging event is not encountered) which it does not have for f-strings (the f-string is evaluated whether or not the event is logged, which is pointless work and for certain log objects could have significant runtime).

We could add other global exceptions if anyone thinks of any, but given how much more readable f strings are, I thought this was an ok trade.